lib/internal/loader/ModuleRequest.js

```diff
@@ -95,14 +95,27 @@ exports.resolve = (specifier, parentURL) => {
                            url.protocol, 'file:');
   }
 
-  if (!preserveSymlinks) {
-    const real = realpathSync(internalURLModule.getPathFromURL(url), {
-      [internalFS.realpathCacheKey]: realpathCache
-    });
-    const old = url;
-    url = internalURLModule.getURLFromFilePath(real);
-    url.search = old.search;
-    url.hash = old.hash;
+  try {
+    if (!preserveSymlinks) {
+      let real;
+      real = realpathSync(internalURLModule.getPathFromURL(url), {
+        [internalFS.realpathCacheKey]: realpathCache
+      });
+      const old = url;
+      url = internalURLModule.getURLFromFilePath(real);
+      url.search = old.search;
+      url.hash = old.hash;
+    }
+    else  {
+      fs.accessSync(internalURLModule.getPathFromURL(url));
+    }
+  }
+  catch (e) {
+    if (e && e.code === 'ENOENT') {
+      e = new Error(`Cannot find module '${specifier}'`);
+      e.code = 'MODULE_NOT_FOUND';
```

Comment:
Is `Error` the correct class, or is there a more precise subclass with a `code`?

Reply:
I've adjusted this to amend the code to the error thrown from C++. If there is a better way to throw the error with the right code in C++ let me know, but I wasn't able to find existing examples for this pattern.